lib/semian.rb

```diff
@@ -150,10 +154,12 @@ def to_s
   #
   # Returns the registered resource.
   def register(name, **options)
-    circuit_breaker = create_circuit_breaker(name, **options)
-    bulkhead = create_bulkhead(name, **options)
-
-    resources[name] = ProtectedResource.new(bulkhead, circuit_breaker)
+    @lock.synchronize do
```

Comment:
Since we register resources so infrequently (just boot), I think it's OK to have the lock even if we disable thread safety and have contention then.